packages/block-library/src/post-content/index.php

```diff
@@ -46,6 +46,11 @@ function render_block_core_post_content( $attributes, $content, $block ) {
 	// so that `the_preview` for the current post can apply.
 	// We force this behavior by omitting the third argument (post ID) from the `get_the_content`.
 	$content = get_the_content( null, false );
+	// Check for nextpage to display page links for paginated posts.
+	$nextpage = has_block( 'nextpage' ) ? wp_link_pages( array( 'echo' => 0 ) ) : false;
+	if ( false !== $nextpage ) {
+		$content = get_the_content() . $nextpage;
+	}
```

Comment:
```suggestion
	if ( has_block( 'core/nextpage' ) ) {
		$content .= wp_link_pages( array( 'echo' => 0 ) );
	}
```

@alberto-marin, a few minor suggestions.

* Let's use the prefix for block name.
* We can also append `wp_link_pages` directly, and there's no need to call `get_the_content` a second time.

Reply:
It looks like there's also a small merge conflict. Once these are resolved, we can merge the PR.

Thanks for your work on this issue!